rl/agents/dqn.py

```diff
@@ -130,9 +130,9 @@ def __init__(self, model, policy=None, test_policy=None, enable_double_dqn=False
             # dueling_type == 'naive'
             # Q(s,a;theta) = V(s;theta) + A(s,a;theta)
             if self.dueling_type == 'avg':
-                outputlayer = Lambda(lambda a: K.expand_dims(a[:, 0], -1) + a[:, 1:] - K.mean(a[:, 1:], keepdims=True), output_shape=(nb_action,))(y)
+                outputlayer = Lambda(lambda a: K.expand_dims(a[:, 0], -1) + a[:, 1:] - K.mean(a[:, 1:], axis=1, keepdims=True), output_shape=(nb_action,))(y)
```

Comment:
I don't really mind changing it since it will do exactly the same thing here, but I don't see how it would be more generic or more consistent. For instance in `backward()` the code uses `actions = np.argmax(q_values, axis=1)` and `q_batch = np.max(target_q_values, axis=1).flatten()` to take the max over actions, not `axis=-1`.

IMO using `-1` here would actually be less generic since if for any reason the tensor `a` had an extra dimension, we would still want to compute the mean over `axis=1` (the action axis) and not this extra dimension.

Anyway, like I said I don't really mind, so please just confirm whether you still prefer the `axis=-1` version and I'll change it accordingly.

Reply:
@delallea, you are correct that -1 would actually make the implementation less generic. Let us go ahead with the current implementation. I will merge this PR instantly. Thank you for your contribution.